File: recursion/permutation.py
# here we implement the permutation function to get all possible permutation
# of string.
def permute(s):
    # Initialize a empty list for storung the result.
    out = []
    # Handling the base case and add the string of len 1 to list.
    if len(s) == 1:
        out = [s]
    else:
        # Loop through the letter of string and enumerate gives the index and
        # letter in for in loop.
        for i,let in enumerate(s):
            # recursively call the permute and loop through each perm to
            # get one permutation.
            for perm in permute(s[:i]+s[i+1:]):
                out += [let + perm]
    return out

print(permute('abc'))

File: recursion/permutation.js
function permute(s) {
    //  Initialize a empty list for storing the result.
    let output  = []
    //  Handling the base case and add the string of len 1 to list.
    if(s.length == 1){
        output.push(s)
    } else {
        //  Loop through the letter of string and  takes the index and
        //  letter in forEach loop.
        for(let i=0; i< s.length; i++){
            //  recursively call the permute and loop through each perm to
            //  get one permutation.
            for (let perm of permute(s.slice(0,i)+s.slice(i+1,s.length))){
                output.push(s[i] + perm)
            }
        }
    }
    return output
}



console.log(permute('abc'))